src/Entity/ListBuilder/SubscriptionListBuilderForDeveloper.php

```diff
@@ -19,309 +19,66 @@
-use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
 
 /**
  * Defines implementation of a subscriptions listing page.
  *
  * @ingroup entity_api
  */
-class SubscriptionListBuilderForDeveloper extends EntityListBuilder implements ContainerInjectionInterface {
+class SubscriptionListBuilderForDeveloper extends SubscriptionListBuilder {
 
   /**
-   * Subscription storage.
-   *
-   * @var \Drupal\apigee_m10n\Entity\Storage\SubscriptionStorageInterface
-   */
-  protected $storage;
-
-  /**
-   * Entity manager service.
-   *
-   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
-   */
-  protected $entityTypeManager;
-
-  /**
-   * Logger service.
-   *
-   * @var \Psr\Log\LoggerInterface
-   */
-  protected $logger;
-
-  /**
-   * Messenger service.
-   *
-   * @var \Drupal\Core\Messenger\MessengerInterface
-   */
-  protected $messenger;
-
-  /**
-   * The user for the listing page.
+   * The developer's user that is used to load subscriptions.
    *
    * @var \Drupal\user\UserInterface
    */
   protected $user;
 
-  /**
-   * SubscriptionListBuilderForDeveloper constructor.
-   *
-   * @param \Drupal\Core\Entity\EntityTypeInterface $entity_type
-   *   Entity type service.
-   * @param \Drupal\Core\Entity\EntityStorageInterface $storage
-   *   Entity storage.
-   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
-   *   Entity type manager service.
-   * @param \Psr\Log\LoggerInterface $logger
-   *   Logger service.
-   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
-   *   Messenger service.
-   */
-  public function __construct(EntityTypeInterface $entity_type, EntityStorageInterface $storage, EntityTypeManagerInterface $entity_type_manager, LoggerInterface $logger, MessengerInterface $messenger) {
-    parent::__construct($entity_type, $storage);
-
-    $this->storage = $storage;
-    $this->entityTypeManager = $entity_type_manager;
-    $this->logger = $logger;
-    $this->messenger = $messenger;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public static function createInstance(ContainerInterface $container, EntityTypeInterface $entity_type) {
-    return new static(
-      $entity_type,
-      $container->get('entity.manager')->getStorage($entity_type->id()),
-      $container->get('entity_type.manager'),
-      $container->get('logger.channel.apigee_m10n'),
-      $container->get('messenger')
-    );
-  }
-
   /**
    * {@inheritdoc}
    */
-  public static function create(ContainerInterface $container) {
-    $entityType = $container->get('entity_type.manager')->getDefinition('subscription');
-    return static::createInstance($container, $entityType);
-  }
+  public function render(UserInterface $user = NULL) {
+    // Return 404 if the user is not set and keep a compatible method signature.
+    if (!($user instanceof UserInterface)) {
+      return new RedirectResponse(Url::fromRoute('system.404'));
```

Comment:
Do we need a redirect here? What about an exception? `throw new NotFoundHttpException();`

Reply:
RedirectResponse requires a string: `return new RedirectResponse(Url::fromRoute('system.404')->toString());`